### PR TITLE
External Audio Recordings with Styling and Error handling

### PR DESCRIPTION
**Pull request by mythilytm**
Primary reviewer:

## Description

- This PR includes changes from other work already completed. Specifically, the following: 
  - https://github.com/techmatters/flex-plugins/pull/1598
  - https://github.com/techmatters/flex-plugins/pull/1620
  - https://github.com/techmatters/flex-plugins/pull/1619
- Styling is one according to mocks: https://marvelapp.com/prototype/e285de8
- Error Handling is similar to transcripts but does not handle both 'permanent' and 'temporary' errors

### Checklist
- [x] Corresponding issue has been opened
- [ ] New tests added
- [n/a] Feature flags added
- [x] Strings are localized
- [x] Tested for chat contacts
- [x] Tested for call contacts

### Related Issues
Fixes # https://tech-matters.atlassian.net/browse/CHI-1348 & https://tech-matters.atlassian.net/browse/CHI-1349
- https://marvelapp.com/prototype/e285de8

### Verification steps
- Start a call and accept it. Save the contact or complete the task
- Search for the newly created contact
- Check the Recording section within this contact and test for the correct behavior of the Chrome audio player. The mocks are here: https://marvelapp.com/prototype/e285de8 
- _Optional_ Ensure external transcripts are still working fine for chat contacts

### CODE REVIEW

File: plugin-hrm-form/src/components/contact/ContactDetailsHome.tsx [hunk 1]
```diff
@@ -240,6 +246,15 @@ const ContactDetailsHome: React.FC<Props> = function ({
       savedContact.details.conversationMedia?.length &&
       (twilioStoredTranscript || externalStoredTranscript),
   );
+
+  const showRecordingSection = Boolean(
+    // featureFlags.enable_voice_recordings &&
```
Comment (mythilytm) [inline plugin-hrm-form/src/components/contact/ContactDetailsHome.tsx:251]: Note to self: enable voice recordings and uncomment this

Comment (GPaoloni) [reply plugin-hrm-form/src/components/contact/ContactDetailsHome.tsx:251]: Please address this before merging

File: plugin-hrm-form/src/components/contact/ContactDetailsHome.tsx [hunk 2]
```diff
@@ -240,6 +246,15 @@ const ContactDetailsHome: React.FC<Props> = function ({
...
+    isVoiceChannel(channel) &&
+      canViewTwilioTranscript &&
```
Comment (GPaoloni) [inline plugin-hrm-form/src/components/contact/ContactDetailsHome.tsx:253]: [viewRecording](https://github.com/techmatters/hrm/blob/d8962375e17b89b0f07f8032032056d9b0cb403c/hrm-domain/hrm-service/permission-rules/dev.json#L25) is the permission we should use here

File: plugin-hrm-form/src/components/contact/MediaSection/RecordingSection.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  return (
+    <div>
+      <audio controls src={voiceRecording} preload="auto" onError={handleFetchAndLoadException}>
+        <track kind="captions" />
+      </audio>
+    </div>
+  );
+};
```
Comment (GPaoloni) [inline plugin-hrm-form/src/components/contact/MediaSection/RecordingSection.tsx:93]: Can we check that we have something in the `voiceRecording` here? And if there is nothing, we just show the button (or any of the above ifs)

**Review by GPaoloni — APPROVED** [review]
Looks good to me!

**Pull request merged**